File: 2023/day-1/puzzle1.cpp
#include <iostream>
#include <fstream>
#include <string>

int main () {
std::ifstream reader; reader.open("test.txt");

if (reader.is_open()) {
    char line;
    while (reader) {
        line = reader.get();
        std::cout << line;
    }
}

}

File: 2023/day-1/puzzle1.py
input = open('./input.txt', 'r')
lines = input.readlines()

sum = 0
for line in lines:
    array = []
    for char in line:
        if char.isdigit():
            array.append(char)
    if array:
        number = int(array[0] + array[-1])
        sum += number
            
print(sum)
